src/execution.jl

```diff
@@ -286,9 +284,9 @@ function generate_benchmark_definition(eval_module, out_vars, setup_vars,
         invocation = :($(Expr(:tuple, out_vars...)) = $(signature))
         core_body = :($(core); $(returns))
     end
-    eval(eval_module, quote
+    eval(quote
```

Comment:
So you want to eval this in `BenchmarkTools`?

Reply:
From my understanding, eval location doesn't actually matter, as long as it's at the top-level of some (any) module. That was my thinking after reading [your comment here](https://github.com/JuliaCI/BenchmarkTools.jl/pull/61#discussion_r121555013), anyway - maybe I misunderstood.

Reply:
My comment wasn't about this case. It's about the two cases above which seems like should be using `invokelatest` instead.
The module you evaluate something in of course has great significance in general and evaluating in the module your code is in is almost always the wrong one (unless it doesn't matter, in which case `eval` is likely not what you want to use, i.e. the case above).
IIUC `core` is user input so it must be evaluated in the module where the user calls this.

Reply:
>The module you evaluate something in of course has great significance in general and evaluating in the module your code is in is almost always the wrong one (unless it doesn't matter, in which case eval is likely not what you want to use, i.e. the case above).

Yeah, I think in this case the evaluation scope doesn't matter, since my only real goal is for it to not match the calling scope. As you mentioned, maybe this indicates that something like `invokelatest` is more appropriate...

> IIUC core is user input so it must be evaluated in the module where the user calls this.

For example,

```julia
julia> for i in 1:10
           @benchmark sin(i)
       end
ERROR: UndefVarError: i not defined
```

...is actually the behavior that I want (and is the current BenchmarkTools behavior). If calling scope matched evaluation scope, I'm worried we might eventually have scope-dependent optimizations that would cause benchmark results to change based on callsite location - which wouldn't be "incorrect", per se, but I think might mislead naive users who aren't aware of what's going on.

Reply:
What about

```julia
module M
using BenchmarkTools
f(i) = i
@benchmark f(1)
end
```

?

Reply:
Ah, good point, now I feel silly XD

Reply:
BTW, by "the case above" I am talking about the case that's actually above this one, i.e. the one mentioned in the comment you linked. Those doesn't seem to be accepting (user generated) expressions as inputs.